### PR TITLE
Add build numbers

### PR DESCRIPTION
**Pull request by tristandl**

### CODE REVIEW

**Comment by aexmachina** [comment]
I can't see how the `$CIRCLE_BUILD_NUM` would be added to the build before it's submitted to iTC...?

**Comment by aexmachina** [comment]
Pushed a small commit with some fixes as I was looking into this. For some reason this branch wasn't up to date with master, but is now.

**Comment by tristandl** [comment]
@aexmachina the env var is used inside an xcode build phase, where the script has always run.

i investigated moving this functionality out of the build phase and into the execute.sh script, but this would mandate modifying the actual plist in the project (at the moment, that is never touched, but we patch the copies of the plist that are injected in to the target) because fastlane/gym doesn't have any way to do this step after a build but before the archive/signing.

File: components/deployment.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ function itunes_connect() {
 
   if [[ $build_number ]]; then
     msg "Tagging build as $build_number"
-    git tag -a build-"$build_number" -m "Added by ios-tools"
+    git tag -a "build-$build_number" -m "Added by ios-tools"
```
Comment (tristandl) [inline components/deployment.sh]: ok, i still don't understand "$VAR" syntax. does it actually put the " character in the command? or is it part of string interpolation?

File: scripts/bundle-version.sh [hunk 2]
```diff
@@ -2,11 +2,11 @@
 
 git=$(sh /etc/profile; which git)
 
-bundle_version="$1"
+bundle_version=$1
```
Comment (tristandl) [inline scripts/bundle-version.sh]: ditto on the "$VAR"! 😢

File: scripts/bundle-version.sh [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 
 if [[ -z $bundle_version ]]; then
   # If not provided as a command line argument, use the number of commits on the current branch
-  bundle_version=$("$git" rev-list HEAD --count)
+  bundle_version=$($git rev-list HEAD --count)
```
Comment (tristandl) [inline scripts/bundle-version.sh]: ha! i didn't write this one at least!

**Comment by aexmachina** [comment]
Need to update the docs to indicate that the _Set Build Number_ build step in Xcode should call `bin/scripts/bundle_version.sh $CIRCLE_BUILD_NUM`. Regarding the bash, let's hop on a quick call and I'll explain 👍

**Comment by tristandl** [comment]
we have a paper doc that has all the project set up steps in it

**Pull request merged**